x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java

```diff
@@ -284,26 +284,30 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
             HttpResponse enrollResponse = null;
             URL enrollNodeUrl = null;
             for (String address : enrollmentToken.getBoundAddress()) {
-                enrollNodeUrl = createURL(new URL("https://" + address), "/_security/enroll/node", "");
-                enrollResponse = client.execute(
-                    "GET",
-                    enrollNodeUrl,
-                    new SecureString(enrollmentToken.getApiKey().toCharArray()),
-                    () -> null,
-                    CommandLineHttpClient::responseBuilder
-                );
-                if (enrollResponse.getHttpStatus() == 200) {
-                    break;
+                try {
+                    enrollNodeUrl = createURL(new URL("https://" + address), "/_security/enroll/node", "");
+                    enrollResponse = client.execute(
+                        "GET",
+                        enrollNodeUrl,
+                        new SecureString(enrollmentToken.getApiKey().toCharArray()),
+                        () -> null,
+                        CommandLineHttpClient::responseBuilder
+                    );
+                    if (enrollResponse.getHttpStatus() == 200) {
+                        break;
+                    }
+                } catch (Exception e) {
+                    terminal.errorPrint(Terminal.Verbosity.NORMAL, "Unable to communicate with the initial node on " + enrollNodeUrl);
```

Comment:
I would suggest we debug log the exception stacktrace; for example, the exception can also be certificate verification exception, or a connection timeout.

Also it is not always connecting with the "initial node". It is connecting with the node that generated the enrollment token.

Reply:
Makes sense, will adjust!

Reply:
printing the stacktrace on the terminal is tricky , would `terminal.errorPrint(Terminal.Verbosity.NORMAL, "Unable to communicate with the node on " + enrollNodeUrl + ". Error was " + e.getMessage());` suffice ?